lib/content.rb

```
@@ -0,0 +1,11 @@
+require 'nokogiri'
+class Content
+  attr_accessor :data,:source
+  def initialize(source)
+    @source = source
+    @data = get_content_from_source
+  end
+  
+  def get_content_from_source
+  end
```

I think you can raise an exception here in case this method is called from an instance of Content class directly.